src/modules/storage/LedgerStorage.ts

```diff
@@ -1646,8 +1953,75 @@ export class LedgerStorage extends Storages {
                             block.txs[tx_idx]
                         );
 
-                        if (block.txs[tx_idx].payload.length > 0)
-                            await save_payload(this, block.merkle_tree[tx_idx], block.txs[tx_idx]);
+                        if (block.txs[tx_idx].payload.length > 0) {
+                            let buffer = SmartBuffer.fromBuffer(block.txs[tx_idx].payload);
+                            let length = VarInt.toNumber(buffer);
+                            let remaining = buffer.remaining();
+                            if (remaining < length) {
+                                await save_payload(this, block.merkle_tree[tx_idx], block.txs[tx_idx]);
+                            } else {
+                                let header = Utils.readBuffer(buffer, length);
+                                switch (header.toString()) {
+                                    case ProposalFeeData.HEADER: {
+                                        let buffer = SmartBuffer.fromBuffer(block.txs[tx_idx].payload);
+                                        let proposal = ProposalFeeData.deserialize(buffer);
+                                        let proposal_id = proposal.proposal_id; // proposal id will be inserted in link below
+                                        let data = await RequestVotera(`http://3.34.7.93:1337/proposals/byid/A11111111`);
+                                        let proposalData: IProposalData = {
```

Comment:
BoaScan should trust and prioritize the information in the payload over the information given by Votera Server.

Reply:
resolved